radix-engine/src/engine/process.rs

```diff
@@ -962,21 +919,37 @@ impl<'r, 'l, L: SubstateStore> Process<'r, 'l, L> {
                     .get_auth(&function);
                 Ok((SNodeState::Bucket(bucket), method_auth.clone()))
             }
-            SNodeRef::Vault(vault_id) => {
+            SNodeRef::BucketRef(bucket_id) => {
```

Comment:
Don't think it makes sense to have both `SNode::Bucket` and `SNode::BucketRef`.  

When we call bucket methods it's always `BucketRef`, except `XXX::func(self)` (which should be calling XXXStatic/XXXBlueprint, where bucket is just an argument).

Reply:
good point

Reply:
Actually, I think we'll want to keep this for now as using BucketStatic won't work given that we'd need to expose arguments to retrieve the method auth. And this may be a useful pattern in the future.

Reply:
But will keep this in mind with further refactors.